### PR TITLE
SiteSelector: display search with less than 6 sites but more hidden ones

### PR DESCRIPTION
**Pull request by mtias**
QoL update so that people have access to search if they have hidden sites but less than 6 visible sites.

cc @aduth 

Test live: https://calypso.live/?branch=update/site-selector-search-display-logic

### CODE REVIEW

File: client/components/site-selector/index.jsx [hunk 1]
```diff
@@ -264,13 +264,12 @@ const SiteSelector = React.createClass( {
 	},
 
 	render() {
+		const hiddenSitesCount = this.props.siteCount - this.props.visibleSiteCount;
 		const selectorClass = classNames( 'site-selector', 'sites-list', {
-			'is-large': this.props.siteCount > 6,
+			'is-large': this.props.siteCount > 6 || hiddenSitesCount > 0,
 			'is-single': this.props.visibleSiteCount === 1
```
Comment (aduth) [inline client/components/site-selector/index.jsx]: Hmm, I don't think we're using this modifier class anywhere, are we?

**Comment by aduth** [comment]
Not necessarily related to the changes here, but when I have a user with two sites, one of which is hidden, I don't have access to the site picker.

If I force the site picker with `?sp` query argument, the search is shown and behaves as I'd expect. In that regard, this LGTM 👍

File: client/components/site-selector/index.jsx (hunk 1 above)
Comment (mtias) [inline client/components/site-selector/index.jsx]: The is-single?

Comment (aduth) [inline client/components/site-selector/index.jsx]: >The is-single?

[Yes](https://github.com/Automattic/wp-calypso/search?utf8=%E2%9C%93&q=.is-single).

**Pull request merged**